### ISSUE TITLE
new `use` and `useOptimistic` hooks in React Native

### ISSUE THREAD

**Issue by subtleGradient** [facebook/react-native#41731]
### New Version

react@18.3.0-canary-c17a27ef4-20231127

### Old Version

react@18.2

### Issue and Reproduction Steps

# TL;DR — bumping React to canary (e.g. react@18.3.0-canary-c17a27ef4-20231127) won't work. Don't try it.

I wanted to use the new 
- `use` (https://react.dev/reference/react/use) and 
- `useOptimistic` (https://react.dev/reference/react/useOptimistic) hooks in React Native.

So, I bumped react to `canary`.

Then I got a runtime error:

> dispatcher.use is not a function (it is undefined)
> dispatcher.useOptimistic is not a function (it is undefined)

Then I tried to figure out what's going on.

I finally stumbled upon: https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Renderer/README.md

Seems like we're going to need to wait until React Native implements these new hooks on the dispatcher side of things before we can use them.

It was very hard to find all this out, so I'm putting this here to help others trying to find out this info.

**Comment by cortinico** [comment]
> TL;DR — bumping React to canary (e.g. react@18.3.0-canary-c17a27ef4-20231127) won't work. Don't try it.

Exactly you can't just bump the React version. It's explained here:
https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Renderer/README.md

You'll have to wait for a stable release of React with the new hooks, and then we'll include them in a stable release of React Native
